docs/rules/no-render-in-setup.md

```diff
@@ -0,0 +1,48 @@
+# Disallow the use of `render` in setup functions (no-render-in-setup)
+
+## Rule Details
+
+This rule disallows the usage of `render` in setup functions (`beforeEach` and `beforeAll`) in favor of moving `render` closer to test assertions.
+
+Examples of **incorrect** code for this rule:
+
+```js
+beforeEach(() => {
+  render(<MyComponent />);
+});
+
+it('Should have foo', () => {
+  expect(screen.getByText('foo')).toBeInTheDocument();
+});
+
+it('Should have bar', () => {
+  expect(screen.getByText('bar')).toBeInTheDocument();
+});
+
+it('Should have baz', () => {
+  expect(screen.getByText('baz')).toBeInTheDocument();
+});
+```
```

Comment:
maybe we should add the `beforeAll` in the examples as it's mentioned too.

Reply:
Good call! Added 👍